src/hyperlight_host/src/mem/mgr.rs

```diff
@@ -617,18 +616,20 @@ impl SandboxMemoryManager<HostSharedMemory> {
 
     /// Writes a guest function call to memory
     #[instrument(err(Debug), skip_all, parent = Span::current(), level= "Trace")]
-    pub(crate) fn write_guest_function_call(&mut self, buffer: &[u8]) -> Result<()> {
-        validate_guest_function_call_buffer(buffer).map_err(|e| {
-            new_error!(
-                "Guest function call buffer validation failed: {}",
-                e.to_string()
-            )
-        })?;
-
+    pub(crate) fn write_guest_function_call(&mut self, function_call: &FunctionCall) -> Result<()> {
+        match function_call.function_call_type() {
+            FunctionCallType::Host => {
+                log_then_return!(
+                    "Tried to serialize a host function call as a guest function call"
+                );
+            }
+            FunctionCallType::Guest => {}
```

Comment:
Nit: maybe move the logic below on the `Guest` match arm or use an `if let FunctionCallType::Host`.